src/xenia/patcher/patcher.cc

```diff
@@ -0,0 +1,217 @@
+/**
+ ******************************************************************************
+ * Xenia : Xbox 360 Emulator Research Project                                 *
+ ******************************************************************************
+ * Copyright 2021 Ben Vanik. All rights reserved.                             *
+ * Released under the BSD license - see LICENSE in the root for more details. *
+ ******************************************************************************
+ */
+#include "xenia/patcher/patcher.h"
+
+#include "xenia/base/cvar.h"
+#include "xenia/base/filesystem.h"
+#include "xenia/base/logging.h"
+
+DEFINE_bool(apply_patches, true, "Enables patching functionality", "General");
+
+namespace xe {
+namespace patcher {
+
+PatchingSystem::PatchingSystem() {
+  is_any_patch_applied_ = false;
+  LoadPatches();
+}
+
+PatchingSystem::~PatchingSystem() {}
+
+void PatchingSystem::LoadPatches() {
+  if (!cvars::apply_patches) {
+    return;
+  }
+
+  const std::filesystem::path patches_directory =
+      filesystem::GetExecutableFolder() / "patches";
```

Comment:
Should we leave it as it is or maybe put it inside ``Documents\Xenia\Patches``?

Reply:
This should be relative to user configuration in some fashion.

Reply:
Yes, relative to the storage root in our terms, which is the "home" directory of Xenia.

Reply:
Though maybe eventually we should support multiple search paths for patches.